### DISCUSSION TITLE
Requesting code review

### DISCUSSION THREAD

**Discussion by pbarni** [community/community#122958] [Programming Help]
### Body

I needed to do a school project, I choose to make a cooking oriented website
this is a static website with little or no javascript (it was not in our task)
the text content is mostly ai generated, and some of the html content was mandatory

aside from that I would really appreciate if you would take a look at it
I would love feedback from just about anything: from design, design choices; the whole document structure tree, html structure, html content choices, css file structure and content structure

Im intrested in what you guys think about the project look as ui / ux, file structure, file readability / maintainability, and code design choises

Im aware that this is not a big project, but if I could fix my mistakes from this, it would be better than nothing. Im looking for opinions of both professionals and starters.

https://github.com/pbarni/cooking_website

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by sondosaabed** [comment]
Hii there!
I have taken a look into it, great job done here! I have these notes that I hope they would be helpful..

- For the svgs file names choose names that actually represent them instead of svg1, svg2,...
- The same goes for the names of the HTML pages instead of page1, page2, page3 have a meaningful name such as indicating their functionality (home, recipes, fun_facts)
- You may also add a conventional footer for the conventional structure of the website pages.
- You may also add comments for better documentation of your code, I think I didn't come across any comment.
- For the images, videos, and svgs you could create a new directory since they are all resources for the website and call it (assets)
- From the eye view, I think the orange color of the logo is a little off of the design and the color palette of the overall design of the pages: 

![image](https://github.com/community/community/assets/65151701/17b87e50-ab85-46fc-ac63-f4c27ecc88d2)
And that's it!
Great job and good luck with your learning journey!
Sondos :)

**Reply by pbarni** [reply]
I had some trouble with the css
a lot of times I had to search for stuff (where did I implement this and that...)
the only 'tool' I used that I created multiple css files so it made the pool I needed to search in smaller

are there any best practices how to structure a css file better?
ofc comments would have helped, but I dont think that it would have solve the problem

**Reply by pbarni** [reply]
also what do you think about the html content itself?
is there anything I could improve on?

also what do you think about the content blocks on the ui? what would you change in the arrangement to make it feel better?

**Reply by sondosaabed** [reply]
> I had some trouble with the css a lot of times I had to search for stuff (where did I implement this and that...) the only 'tool' I used that I created multiple css files so it made the pool I needed to search in smaller
> 
> are there any best practices how to structure a css file better? ofc comments would have helped, but I dont think that it would have solve the problem

I have a suggestion about the styling it could be more centralized such this architecture:
- https://github.com/sondosaabed/Socket-WebServer-Networking-Project/tree/main/Part-3/src/css

![image](https://github.com/community/community/assets/65151701/270b9646-2151-4682-b09e-514fe3816e52)

the fonts, the colors, image style each in one styling sheet.


> also what do you think about the html content itself? is there anything I could improve on?
> 
> also what do you think about the content blocks on the ui? what would you change in the arrangement to make it feel better?

The way that the blocks are arranged we call it layout, there are different layouts that are trendy you could use. The main thing is to have the 7 sections or blocks (header, navigation, body, footer....) when you have these sections you build trust with the user because this is what they are used to.

For your html yes I forgot to mention this, instead of using the `<div>` start using semantic html, there are tags like <`header>, <body>, <section>`. These are used instead of div a better practice for (SEO) search engine optimization.
Like this image show:

![image](https://github.com/community/community/assets/65151701/cf67eb3f-dd44-48a2-9770-04f80607e162)


Look into these topics:
- layouts (css)
- semantic html

I remembered there is two links one for html and one for css that checks them and how to improve them let me look for them for you.

**Reply by sondosaabed** [reply]
Here is the css validator
https://jigsaw.w3.org/css-validator/

and here is the html validator:
https://validator.w3.org/

these could help in learning were to improve and what not to do

**Reply by pbarni** [reply]
as every website different, they have different needs
should you use every tag, always?
what if your website do not need an aside?
this aside is confusing me, maybe its about this graph you showed me
in the graph it is aside of main, not an article... im not sure

**Comment by virajbhutada** [comment]
Hey there! I checked out your cooking website project on GitHub. It's great that you're seeking feedback to improve it! I'll take a look at the UI/UX, file structure, readability, and code design choices. Let's jump in and see how we can enhance your project together.
